mm-bot/src/services/ethereum.py

```diff
@@ -31,6 +31,20 @@ def get_latest_block() -> int:
     logger.error(f"[-] Failed to get block number from all nodes")
 
 
+def get_is_used_order(order_id, recipient_address, amount) -> bool:
+    is_used_index = 2
+    order_data = Web3.solidity_keccak(['uint256', 'uint256', 'uint256'],
+                                      [order_id, int(recipient_address, 0), amount])
+    for contract in contracts:
```

Comment:
Why is it iterating the contracts array? Also there is only one contract. The fallback function is just a function from the same contract. We use that instead of calling the Herodotus API.

Reply:
There are two contracts because each one uses the web3 clients with Alchemy and Infura. So
main_w3_client uses infura RPC -> we create a contract instance using this client
fallback_w3_client uses alchemy RPC -> we a create a contract instance using this client

Its the same contract, with differents clients using different RPCs to provide fault tolerance

The herodotus API is not involved here

Reply:
Oh ok. Couldn't we changed the name then? It can be misleading calling this variable contracts.
This variables are the instances of the variable that allows us to interact with our contract with a specific RPC right?